circleci.tf

```diff
@@ -2,461 +2,529 @@
+}
+
+variable "services_ami" {
+  description = "Override AMI lookup with provided AMI ID"
+  default     = ""
 }
 
 variable "services_instance_type" {
-    description = "instance type for the centralized services box.  We recommend a c4 instance"
-    default = "c4.2xlarge"
+  description = "instance type for the centralized services box.  We recommend a c4 instance"
+  default     = "c4.2xlarge"
 }
 
 variable "builder_instance_type" {
-    description = "instance type for the builder machines.  We recommend a r3 instance"
-    default = "r3.2xlarge"
+  description = "instance type for the builder machines.  We recommend a r3 instance"
+  default     = "r3.2xlarge"
 }
 
 variable "max_builders_count" {
-    description = "max number of builders"
-    default = "2"
+  description = "max number of builders"
+  default     = "2"
 }
 
 variable "prefix" {
-    description = "prefix for resource names"
-    default = "circleci"
+  description = "prefix for resource names"
+  default     = "circleci"
+}
+
+variable "enable_ansible_provisioning" {
+  description = "Enables Ansible provisioning of the Services box for automatic / no-touch installation / configuration."
+  default     = 0
+}
+
+variable "ansible_extra_vars" {
+  type    = "map"
+  default = {}
+}
+
+variable "services_delete_on_termination" {
+  description = "Configures AWS to delete the ELB volume for the Services box upon instance termination."
+  default     = "false"
+}
+
+variable "enable_route" {
+  description = "enable creating a Route53 route for the Services box"
+  default     = 0
+}
+
+variable "route_name" {
+  description = "Route name to configure for Services box"
+  default     = ""
+}
+
+variable "route_zone_id" {
+  description = "Zone to configure route in"
+  default     = ""
 }
 
 data "aws_subnet" "subnet" {
   id = "${var.aws_subnet_id}"
 }
 
+data "template_file" "services_user_data" {
+  template = "${file("templates/services_user_data.tpl")}"
+
+  vars {
+    circle_secret_passphrase = "${var.circle_secret_passphrase}"
+    sqs_queue_url            = "${aws_sqs_queue.shutdown_queue.id}"
+    s3_bucket                = "${aws_s3_bucket.circleci_bucket.id}"
+    aws_region               = "${var.aws_region}"
+    subnet_id                = "${var.aws_subnet_id}"
+    vm_sg_id                 = "${aws_security_group.circleci_vm_sg.id}"
+  }
+}
+
+data "template_file" "builders_user_data" {
+  template = "${file("templates/builders_user_data.tpl")}"
+
+  vars {
+    services_private_ip      = "${aws_instance.services.private_ip}"
+    circle_secret_passphrase = "${var.circle_secret_passphrase}"
+  }
+}
+
+data "template_file" "circleci_policy" {
+  template = "${file("templates/circleci_policy.tpl")}"
+
+  vars {
+    bucket_arn    = "${aws_s3_bucket.circleci_bucket.arn}"
+    sqs_queue_arn = "${aws_sqs_queue.shutdown_queue.arn}"
+  }
+}
+
+data "template_file" "shutdown_queue_role_policy" {
```

Comment:
Good to know. I was just trying to clean up the file and moved blocks of strings into templates.